src/runner_test.go

```diff
@@ -35,55 +30,91 @@ vars:
         echo "$RHC_WORKER_FOO $RHC_WORKER_BAR!"
     content_vars:
         FOO: Hello
-        BAR: World`)
-	expectedResult = "Hello World!\n"
-	result = processSignedScript(yamlData)
-	if result != expectedResult {
-		t.Errorf("Expected %q, but got %q", expectedResult, result)
+        BAR: World`),
+			expectedResult: "Hello World!\n",
+		},
+		{
+			name:       "verification enabled, invalid signature = error msg returned",
+			verifyYAML: true,
+			yamlData: []byte(`
+vars:
+    _insights_signature: "invalid-signature"
+    _insights_signature_exclude: "/vars/insights_signature,/vars/content_vars"
+    content: |
+        #!/bin/sh
+        echo "$RHC_WORKER_FOO $RHC_WORKER_BAR!"
+    content_vars:
+        FOO: Hello
+        BAR: World`),
+			expectedResult: "Signature of yaml file is invalid",
+		},
 	}
 
-	// FIXME: This is false success because verification fails on missing insighs-client
-	// Test case 3: verification enabled, invalid signature = error msg returned
-	shouldVerifyYaml = true
-	shouldDoInsightsCoreGPGCheck = true
-	expectedResult = "Signature of yaml file is invalid"
-	result = processSignedScript(yamlData)
-	if result != expectedResult {
-		t.Errorf("Expected %q, but got %q", expectedResult, result)
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			shouldVerifyYaml := tc.verifyYAML
+			shouldDoInsightsCoreGPGCheck := tc.verifyYAML // Assume the same value for simplicity
+			temporaryWorkerDirectory := "test-dir"
+			config = &Config{
+				VerifyYAML:               &shouldVerifyYaml,
+				TemporaryWorkerDirectory: &temporaryWorkerDirectory,
+				InsightsCoreGPGCheck:     &shouldDoInsightsCoreGPGCheck,
+			}
+
+			defer os.RemoveAll(temporaryWorkerDirectory)
+
+			result := processSignedScript(tc.yamlData)
+			if result != tc.expectedResult {
+				t.Errorf("Expected %q, but got %q", tc.expectedResult, result)
+			}
+		})
 	}
 }
 
 func TestVerifyYamlFile(t *testing.T) {
-	shouldVerifyYaml := false
-	shouldDoInsightsCoreGPGCheck := false
-
-	config = &Config{
-		VerifyYAML:           &shouldVerifyYaml,
-		InsightsCoreGPGCheck: &shouldDoInsightsCoreGPGCheck,
-	}
-	// Test case 1: verification disabled
-	expectedResult := true
-	result := verifyYamlFile([]byte{})
-	if result != expectedResult {
-		t.Errorf("Expected %v, but got %v", expectedResult, result)
+	testCases := []struct {
+		name           string
+		verifyYAML     bool
+		yamlData       []byte
+		expectedResult bool
+	}{
+		{
+			name:           "verification disabled",
+			verifyYAML:     false,
+			yamlData:       []byte{},
+			expectedResult: true,
+		},
+		// FIXME: This should succedd but now verification fails on missing insighs-client
```

Comment:
```suggestion
		// FIXME: This should succeed but now verification fails on missing insighs-client
```

Reply:
Also, I'm in favour of mocking the insights-client call... I know it might not be the best approach ever, but in this case, we can't rely of it being on the system, and more especially, of it doing the correct verification in our unit tests. 

This is something I would expect a integration test to do, rather than our unit testing.

Reply:
Alright, I initially though it will be harder but got an idea today, it is mocked - added in separate commit